### PR TITLE
feat: add order and type for uuidtobin and binToUuid

### PR DESCRIPTION
**Pull request by zgoby**
I pull request for adding type and add the order param by the way

### CODE REVIEW

File: lib/knex-builder/FunctionHelper.js [hunk 1]
```diff
@@ -14,25 +14,30 @@ class FunctionHelper {
     return this.client.raw('CURRENT_TIMESTAMP');
   }
 
-  uuidToBin(uuid) {
+  uuidToBin(uuid, flag = true) {
```
Comment (OlivierCavadenti) [inline lib/knex-builder/FunctionHelper.js:17]: flag => ordered

File: lib/knex-builder/FunctionHelper.js [hunk 2]
```diff
@@ -14,25 +14,30 @@ class FunctionHelper {
...
-    const buf = new Buffer(bin, 'hex');
-    return [
-      buf.toString('hex', 4, 8),
-      buf.toString('hex', 2, 4),
-      buf.toString('hex', 0, 2),
-      buf.toString('hex', 8, 10),
-      buf.toString('hex', 10, 16),
-    ].join('-');
+  
+  binToUuid(bin, flag = true) {
```
Comment (OlivierCavadenti) [inline lib/knex-builder/FunctionHelper.js:24]: flag => ordered

File: lib/knex-builder/FunctionHelper.js (hunk 1 above)
Comment (OlivierCavadenti) [inline lib/knex-builder/FunctionHelper.js:17]: Can you add some comment to say what is the goal of this flag ? I dont know why is necessary.

**Review by OlivierCavadenti — COMMENTED** [review]
Add unit and integration tests.
Missing doc : https://github.com/knex/documentation/

File: lib/knex-builder/FunctionHelper.js (hunk 1 above)
Comment (zgoby) [reply lib/knex-builder/FunctionHelper.js:17]: it just give an option of UUID_TO_BIN(UUID(), false) for mysql  to ensure that the logic of the inserted data is consistent with old data

Comment (OlivierCavadenti) [reply lib/knex-builder/FunctionHelper.js]: For other reviewers, the doc in mysql about swap flag : https://dev.mysql.com/doc/refman/8.0/en/miscellaneous-functions.html#function_uuid-to-bin

File: test/integration2/query/misc/additional.spec.js [hunk 3]
```diff
@@ -394,6 +394,39 @@ describe('Additional', function () {
           const binary = knex.fn.uuidToBin(originalUuid);
           const uuid = knex.fn.binToUuid(binary);
           expect(uuid).to.equal(originalUuid);
+          const binaryUnoder = knex.fn.uuidToBin(originalUuid, false);
+          const uuidUnoder = knex.fn.binToUuid(binary, false);
```
Comment (OlivierCavadenti) [inline test/integration2/query/misc/additional.spec.js:398]: binaryUnoder => binaryUnorder
uuidUnoder=> uuidUnorder

File: test/integration2/query/misc/additional.spec.js [hunk 4]
```diff
@@ -394,6 +394,39 @@ describe('Additional', function () {
...
+          expect(binaryUnoder).to.equal(uuidUnoder);
+        });
+
+        it('should insert binary uuid and retrieve it', async () => {
```
Comment (OlivierCavadenti) [inline test/integration2/query/misc/additional.spec.js:402]: should insert binary uuid and retrieve it with not ordered uuid data

File: lib/knex-builder/FunctionHelper.js (hunk 1 above)
Comment (zgoby) [reply lib/knex-builder/FunctionHelper.js]: yep, i think true  as default is better

File: test/integration2/query/misc/additional.spec.js [hunk 5]
```diff
@@ -394,6 +394,39 @@ describe('Additional', function () {
           const binary = knex.fn.uuidToBin(originalUuid);
           const uuid = knex.fn.binToUuid(binary);
           expect(uuid).to.equal(originalUuid);
+          const binaryUnorder = knex.fn.uuidToBin(originalUuid, false);
+          const uuidUnorder = knex.fn.binToUuid(binary, false);
+          expect(binaryUnorder).to.equal(uuidUnorder);
```
Comment (OlivierCavadenti) [inline test/integration2/query/misc/additional.spec.js:399]: Fix this test : 

```js
          const binaryUnorder = knex.fn.uuidToBin(originalUuid, false);
          const uuidUnorder = knex.fn.binToUuid(binaryUnorder, false);
          expect(uuidUnorder).to.equal(originalUuid);
```

**Review by OlivierCavadenti — CHANGES_REQUESTED** [review]

**Review by OlivierCavadenti — APPROVED** [review]

**Comment by kibertoad** [comment]
Released in 1.0.2

**Pull request merged**